### PR TITLE
Refactoring, making the recipe e2e including data generation

### PR DESCRIPTION
**Pull request by popcornell**
Adding more step by step installation guide. 
Also I will refactor the launchers to add datapreparation

### CODE REVIEW

File: scripts/chime8/pipeline/confs/chime_config.yaml [hunk 1]
```diff
@@ -119,14 +118,15 @@ diarizer:
...
   clustering:
     parameters:
+      frame_vad_threshold: 0.8 #${frame_vad_threshold}
       oracle_num_speakers: False # If True, use num of speakers value provided in manifest file.
       max_num_speakers: ${max_num_spks} # Max number of speakers for each recording. If an oracle number of speakers is passed, this value is ignored.
-      min_num_speakers: 2 # Max number of speakers for each recording. If an oracle number of speakers is passed, this value is ignored.
+      min_num_speakers: 2 # Min number of speakers for each recording. If an oracle number of speakers is passed, this value is ignored.
       enhanced_count_thres: 80 # If the number of segments is lower than this number, enhanced speaker counting is activated.
-      max_rp_threshold: 0.06
+      max_rp_threshold: 0.9
```
Comment (popcornell) [inline scripts/chime8/pipeline/confs/chime_config.yaml:127]: these needs to be tuned

File: scripts/chime8/pipeline/local/gss/chime7_enhancers.py [hunk 2]
```diff
@@ -22,8 +22,8 @@
 import soundfile as sf
 import torch
 import torchaudio
-from gss.core import Activity
-from gss.utils.data_utils import GssDataset, create_sampler
+from .core.activity import Activity
```
Comment (popcornell) [inline scripts/chime8/pipeline/local/gss/chime7_enhancers.py:25]: removfed dependencies from cupy gss.

File: scripts/chime8/installers/install_c8_dependencies.sh [hunk 3]
```diff
@@ -0,0 +1,5 @@
+pip install optuna
+pip install git+https://github.com/chimechallenge/chime-utils
+pip install git+https://github.com/kpu/kenlm
+pip install cmake>=3.18
+sh ./installers/run_install_lm.sh ${PWD}/../../../NeMo
```
Comment (tango4j) [inline scripts/chime8/installers/install_c8_dependencies.sh]: `NeMo` -> `C8DASR-Baseline-NeMo`
Path name should be updated.
Please test this script again.

File: scripts/chime8/installers/install_c8_dependencies.sh [hunk 4]
```diff
@@ -0,0 +1,5 @@
+pip install optuna
+pip install git+https://github.com/chimechallenge/chime-utils
+pip install git+https://github.com/kpu/kenlm
+pip install cmake>=3.18
+sh ./installers/run_install_lm.sh ${PWD}/../../../C8DASR-Baseline-NeMo
```
Comment (stevehuang52) [inline scripts/chime8/installers/install_c8_dependencies.sh]: should remove the `./installers` prefix from the path if running under `./installers`?

File: scripts/chime8/pipeline/launchers/launch_inference_optuna.sh [hunk 5]
```diff
@@ -19,7 +19,7 @@ MSDD_MODEL_PATH=${CHECKPOINTS_DIR}/msdd_v2_PALO_bs6_a003_version6_e53.ckpt
 SCENARIOS="[mixer6]"
 SUBSETS="[dev]"  # can only optimize one subset
 
-SCRIPT_NAME=${NEMO_ROOT}/scripts/chime7/pipeline/run_full_pipeline_optuna.py
+SCRIPT_NAME=${NEMO_ROOT}/scripts/chime7/pipeline/inference_optuna.py
```
Comment (stevehuang52) [inline scripts/chime8/pipeline/launchers/launch_inference_optuna.sh]: `inference_optuna.py` file not found? is it `tune_all.py`?

File: scripts/chime8/installers/install_c8_dependencies.sh (hunk 4 above)
Comment (popcornell) [reply scripts/chime8/installers/install_c8_dependencies.sh]: yes good catch

File: nemo/collections/asr/models/clustering_multi_ch_diarizer.py [hunk 6]
```diff
@@ -617,7 +617,7 @@ def _run_clustering(self, embeddings, time_stamps, vad_probs, scale_mapping):
             multiscale_weights=self._diarizer_params.speaker_embeddings.parameters.multiscale_weights,
             multiscale_dict=self.multiscale_args_dict['scale_dict'],
             verbose=self.verbose,
-            vad_threshold=self._diarizer_params.vad.parameters.frame_vad_threshold,
+            vad_threshold=self._diarizer_params.clustering.parameters.frame_vad_threshold,
```
Comment (stevehuang52) [inline nemo/collections/asr/models/clustering_multi_ch_diarizer.py]: what's the motivation of this change?

Comment (popcornell) [reply nemo/collections/asr/models/clustering_multi_ch_diarizer.py]: I think that maybe it is good to have stricter VAD for embedding extraction ? 
Because right now the speaker counting is off. Maybe we don't need that because MSDD will refine the predictions nonetheless.

Comment (stevehuang52) [reply nemo/collections/asr/models/clustering_multi_ch_diarizer.py]: It seems that changing the name from `_diarizer_params.vad.parameters.frame_vad_threshold` to `_diarizer_params.clustering.parameters.frame_vad_threshold` won't affect anything, as `frame_vad_threshold` is only used here....? @tango4j am I understanding it correctly?

File: scripts/chime8/pipeline/launchers/launch_inference_optuna.sh (hunk 5 above)
Comment (popcornell) [reply scripts/chime8/pipeline/launchers/launch_inference_optuna.sh]: I changed it ! Thanks a lot
